klippy/extras/sht4x.py

```diff
@@ -0,0 +1,262 @@
+        self.measurement_lock = threading.Lock()
+        self._callback = None
+        self.serial_number = None
+
+        # Register sensor
+        self.printer.add_object("sht4x " + self.name, self)
+        self.printer.register_event_handler("klippy:connect",
+                                           self.handle_connect)
+
+    def handle_connect(self):
+        """Initialize sensor"""
+        try:
+            self.reset()
+            # Add a small delay after reset
+            self.reactor.pause(self.reactor.monotonic() + 0.01)  # 10ms
+
+            # Try to read serial number
+            self.serial_number = self.get_serial_number()
+            if self.serial_number:
+                logging.info("SHT4X: Sensor serial number: 0x%08X",
+                           self.serial_number)
+
+            # Test measurement
+            if self._test_measurement():
+                self.sensor_ready = True
+                logging.info("SHT4X: Sensor ready")
+                self.sample_timer = self.reactor.register_timer(
+                    self.sample_sensor)
+                self.reactor.update_timer(self.sample_timer,
+                                        self.reactor.NOW)
+            else:
+                logging.error("SHT4X: Sensor test failed")
+        except Exception as e:
+            logging.error("SHT4X: Init failed: %s" % str(e))
+
+    def _test_measurement(self):
+        """Quick measurement test"""
+        try:
+            recv = self.get_measurements()
+            if len(recv) != 6:
+                logging.error("SHT4X: Invalid data length: %d", len(recv))
+                return False
+            if not self._validate_crc(recv):
+                logging.error("SHT4X: CRC validation failed")
+                return False
+            return True
+        except Exception as e:
+            logging.error("SHT4X: Test measurement failed: %s", str(e))
+            return False
+
+    def _validate_crc(self, data):
+        """Validate CRC for both temp and humidity"""
+        temp_crc = data[2]
+        humidity_crc = data[5]
+        return (temp_crc == self._crc8(data[0:2]) and
+                humidity_crc == self._crc8(data[3:5]))
+
+    def setup_minmax(self, min_temp, max_temp):
+        pass
+
+    def setup_callback(self, cb):
+        self._callback = cb
+
+    def sample_sensor(self, eventtime):
+        """Main sampling loop"""
+        if not self.sensor_ready:
+            return eventtime + REPORT_TIME
+
+        try:
+            with self.measurement_lock:
+                recv = self.get_measurements()
+
+                if len(recv) != 6 or not self._validate_crc(recv):
+                    raise Exception("Invalid data or CRC failed")
+
+                # Convert temperature
+                raw_temp = (recv[0] << 8) | recv[1]
+                temp = -45.0 + 175.0 * raw_temp / 65535.0
+
+                # Check if temperature is in reasonable range
+                if -50 <= temp <= 150:
```

Comment:
This one looks weird to me,
You have a formula above that would not be able to return any 16-bit value outside the -45 and 130.